GoodThingsComein3s/ViewControllers/SignUpLoginViewController.h

```diff
@@ -6,11 +6,20 @@
 //
 
 #import <UIKit/UIKit.h>
+#import <Parse/Parse.h>
+#import "Restaurant.h"
 
 NS_ASSUME_NONNULL_BEGIN
 
+@protocol SignUpLoginViewControllerDelegate
+-(void) addLikedRestaurantToUser:(PFUser *)currUser restaurant:(Restaurant *)restaurant;
```

Comment:
A class should only use delegate methods to notify its delegate that certain actions occured, in this case, that the signupLoginVC was dismissed. It's up to the delegate to do what it wants with that notification, in this case, it will add the liked restaurant to current user's database. So in this case, this method should be called something like:

`- (void)signUpLoginViewControllerDidDismissForUser:(PFUser *)currentUser withLikedRestaurant:(Restaurant *)restaurant;`

Also another reminder to be more descriptive with delegate method names. It makes it easier from just reading to identify what is a delegate method and isn't.